app/routes/surveys/AddSurveyRoute.js

```diff
@@ -1,31 +1,85 @@
 import { connect } from 'react-redux';
 import { compose } from 'redux';
-import { addSurvey, deleteSurvey } from '../../actions/SurveyActions';
+import {
+  addSurvey,
+  deleteSurvey,
+  fetchTemplate
+} from '../../actions/SurveyActions';
 import SurveyEditor from './components/SurveyEditor/SurveyEditor';
 import { LoginPage } from 'app/components/LoginForm';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
 import { push } from 'react-router-redux';
-import moment from 'moment-timezone';
 import { initialQuestion } from './components/SurveyEditor/SurveyEditor';
+import prepare from 'app/utils/prepare';
+import { selectSurveyTemplate } from 'app/reducers/surveys';
+import { fetchEvent } from 'app/actions/EventActions';
+import { selectEventById } from 'app/reducers/events';
+import { defaultActiveFrom } from './utils';
 
-const time = (hours, minutes) =>
-  moment()
-    .startOf('day')
-    .add({ hours, minutes })
-    .toISOString();
+const loadData = (props, dispatch) => {
+  const { template, event } = props.location.query;
+  if (event) {
+    return dispatch(fetchEvent(event)).then(
+      result =>
+        result.success &&
```

Comment:
do you need the `result.success` here? wouldn't it throw if the result wasn't successful?

Reply:
Okay? I figured since `result` includes a boolean `success` there were cases where it was false I'd have to handle, but I wouldn't be surprised if that's taken care of in middleware or something.